packages/nextjs/src/config/loaders/ast.ts

```diff
@@ -320,3 +328,169 @@ export function removeComments(ast: AST): void {
+  const identifiers: string[] = [];
+
+  if (Identifier.check(restElement.argument)) {
+    identifiers.push(restElement.argument.name);
+  }
+
+  return identifiers;
+}
+
+/**
+ * Extracts all identifiers from an ArrayPattern within a destructured named export declaration
+ * statement ("export const [val] = [1]").
+ *
+ * This function recursively calls itself and `getExportIdentifiersFromObjectPattern` since destructuring assignments
+ * can be deeply nested with objects and arrays.
+ *
+ * Example:
+ * export const [{ foo: name1 }, [{ bar: [name2]}, name3]] = [{ foo: 1 }, [{ bar: [2] }, 3]];
+ *   --> ["name1", "name2", "name3"]
+ */
+function getExportIdentifiersFromArrayPattern(arrayPattern: jscsTypes.ArrayPattern): string[] {
+  const identifiers: string[] = [];
+
+  arrayPattern.elements.forEach(element => {
+    if (Identifier.check(element)) {
+      identifiers.push(element.name);
+    } else if (ObjectPattern.check(element)) {
+      identifiers.push(...getExportIdentifiersFromObjectPattern(element));
+    } else if (ArrayPattern.check(element)) {
+      identifiers.push(...getExportIdentifiersFromArrayPattern(element));
+    } else if (RestElement.check(element)) {
+      identifiers.push(...getExportIdentifiersFromRestElement(element));
+    }
+  });
+
+  return identifiers;
+}
+
+/**
+ * Grabs all identifiers from an ObjectPattern within a destructured named export declaration
+ * statement ("export const { val: name } = { val: 1 }").
+ *
+ * This function recursively calls itself and `getExportIdentifiersFromArrayPattern` since destructuring assignments
+ * can be deeply nested with objects and arrays.
+ *
+ * Example:
+ * export const { foo: [name1], bar: { baz: [{ quux: name2 }], ...name3 }} = { foo: [1], bar: { baz: [{ quux: 3 }]} };
+ *   --> ["name1", "name2", "name3"]
+ */
+function getExportIdentifiersFromObjectPattern(objectPatternNode: jscsTypes.ObjectPattern): string[] {
+  const identifiers: string[] = [];
+
+  objectPatternNode.properties.forEach(property => {
+    if (Property.check(property)) {
+      if (Identifier.check(property.value)) {
+        identifiers.push(property.value.name);
+      } else if (ObjectPattern.check(property.value)) {
+        identifiers.push(...getExportIdentifiersFromObjectPattern(property.value));
+      } else if (ArrayPattern.check(property.value)) {
+        identifiers.push(...getExportIdentifiersFromArrayPattern(property.value));
+      } else if (RestElement.check(property.value)) {
+        identifiers.push(...getExportIdentifiersFromRestElement(property.value));
+      }
+    } else if (RestElement.check(property)) {
+      identifiers.push(...getExportIdentifiersFromRestElement(property));
+    }
+  });
+
+  return identifiers;
+}
+
+/**
+ * Given the AST of a file, this function extracts all named exports from the file.
+ *
+ * @returns a list of deduplicated identifiers.
+ */
+export function getExportIdentifiers(ast: AST): string[] {
```

Comment:
```suggestion
export function getExportIdentifierNames(ast: AST): string[] {
```
Since other function names mentioning "identifiers" are referring to `Identifiers`, I think we should make it clear that this is different.

Reply:
Good suggestion: aef96355a90c0112e9daddb0b73345bb7ba655b6